app/models/Collection.js

```diff
@@ -133,11 +149,11 @@ class Collection extends BaseModel {
       if (data.collectionId === this.id) this.fetch();
     });
     this.on(
-      'collections.update',
-      (data: { id: string, collection: Collection }) => {
-        // FIXME: calling this.updateData won't update the
```

Comment:
Addressed by using `withRouter` around `SidebarLink` to ensure re-render happens. 

https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/guides/blocked-updates.md